File: ArabPro/src/main/java/com/Backend/ArabPro/models/Task.java
package com.Backend.ArabPro.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
@Entity
@NoArgsConstructor
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String priority;
    private String libelle;
    private String description;
    private Date date;
    
    @Enumerated(EnumType.STRING)
    private TaskStatus status;


    @ManyToOne
    @JoinColumn(name = "employee_id")
    private User employee;

    @ManyToOne
    @JoinColumn(name = "project_id")
    private Project project;



}

File: ArabProFront/src/app/models/Task.ts
import { Project } from "./Project";
import { User } from "./User";

export class Task{
    id!:number;
    priority!:string;
    libelle!:string;
    description!:string;
    status!:TaskStatus;
    date!:Date;

    employee: User | undefined;

    project: Project | undefined;
}
export enum TaskStatus{
    NEW='NEW',TODO ='TODO', IN_PROGRESS = 'IN_PROGRESS', REVIEW = 'REVIEW' ,DONE = 'DONE'
}